cmd/gnodev/mod.go

```diff
@@ -12,38 +12,62 @@ import (
 	"github.com/gnolang/gno/pkgs/gnolang/gnomod"
 )
 
-func newModCmd() *commands.Command {
-	return commands.NewCommand(
+type modDownloadCfg struct {
+	remote string
+}
+
+var defaultModDownloadOptions = &modDownloadCfg{
+	remote: "test3.gno.land:36657",
+}
+
+func newModCmd(io *commands.IO) *commands.Command {
+	cmd := commands.NewCommand(
 		commands.Metadata{
 			Name:       "mod",
 			ShortUsage: "mod <command>",
 			ShortHelp:  "Manage gno.mod",
 		},
 		commands.NewEmptyConfig(),
+		commands.HelpExec,
+	)
+
+	cmd.AddSubCommands(
+		newModDownloadCmd(io),
+	)
+
+	return cmd
+}
+
+func newModDownloadCmd(io *commands.IO) *commands.Command {
+	cfg := &modDownloadCfg{}
+
+	return commands.NewCommand(
+		commands.Metadata{
+			Name:       "download",
+			ShortUsage: "download [flags]",
+			ShortHelp:  "Download modules to local cache",
+		},
+		cfg,
 		func(_ context.Context, args []string) error {
-			return execMod(args)
+			return execModDownload(cfg, args, io)
 		},
 	)
 }
 
-func execMod(args []string) error {
-	if len(args) != 1 {
-		return flag.ErrHelp
-	}
+func (c *modDownloadCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.StringVar(
+		&c.remote,
+		"remote",
+		defaultModDownloadOptions.remote,
+		"remote for fetching gno modules",
+	)
+}
 
-	switch args[0] {
-	case "download":
-		if err := runModDownload(); err != nil {
-			return fmt.Errorf("mod download: %w", err)
-		}
-	default:
-		return fmt.Errorf("invalid command: %s", args[0])
+func execModDownload(cfg *modDownloadCfg, args []string, io *commands.IO) error {
```

Comment:
Why did you pass in the IO if you do not use it anywhere yet?

Reply:
I was planning to add `verbose` flag in this PR. But some more work needs to be done before adding that. So dropped the idea. Can we keep it for future?